source/java/spring-boot-web/tutorial.html

```diff
@@ -0,0 +1,845 @@
+that the Spring Security <code class="docutils literal"><span class="pre">hasPermission</span></code> clause will respond to. Note, this does not preclude you in any way from having other custom data.</p>
+<p>Let&#8217;s take a look at the updated <code class="docutils literal"><span class="pre">HelloService</span></code>:</p>
+<div class="highlight-java"><table class="highlighttable"><tr><td class="linenos"><div class="linenodiv"><pre>1
+2
+3
+4
+5
+6
+7
+8</pre></div></td><td class="code"><div class="highlight"><pre><span class="nd">@Service</span>
+<span class="kd">public</span> <span class="kd">class</span> <span class="nc">HelloService</span> <span class="o">{</span>
+<span class="hll">    <span class="nd">@PreAuthorize</span><span class="o">(</span><span class="s">&quot;hasRole(@Roles.USER) and hasPermission(&#39;say&#39;, &#39;hello&#39;)&quot;</span><span class="o">)</span>
+</span>    <span class="kd">public</span> <span class="n">String</span> <span class="nf">sayHello</span><span class="o">(</span><span class="n">Account</span> <span class="n">account</span><span class="o">)</span> <span class="o">{</span>
+        <span class="k">return</span> <span class="s">&quot;Hello, &quot;</span> <span class="o">+</span> <span class="n">account</span><span class="o">.</span><span class="na">getGivenName</span><span class="o">()</span> <span class="o">+</span>
+            <span class="s">&quot;. You have the required permissions to access this restricted resource.&quot;</span><span class="o">;</span>
+    <span class="o">}</span>
+<span class="o">}</span>
+</pre></div>
+</td></tr></table></div>
+<p>Notice that in addition to the <code class="docutils literal"><span class="pre">hasRole</span></code> clause, we now have added: <code class="docutils literal"><span class="pre">hasPermission('say',</span> <span class="pre">'hello')</span></code>. We are saying that
+in addition to the user being in the group identified by <code class="docutils literal"><span class="pre">&#64;Roles.USER</span></code>, they must also have the permission to say hello.</p>
+<p>This is connected to Stormpath by having the following custom data present:</p>
+<div class="highlight-javascript"><div class="highlight"><pre><span class="p">{</span>
+    <span class="s2">&quot;springSecurityPermissions&quot;</span><span class="o">:</span> <span class="p">[</span><span class="s2">&quot;say:hello&quot;</span><span class="p">]</span>
+<span class="p">}</span>
+</pre></div>
+</div>
+<p><code class="docutils literal"><span class="pre">springSecurityPermissions</span></code> is the special key I talked about above. Its value is an array of strings each of which
+conforms to the following format: <code class="docutils literal"><span class="pre">target:permission</span></code>. In this case, the target is <em>say</em> and the permission is <em>hello</em>.</p>
```

Comment:
Permissions can actually be more flexible than this. Please see the concept of WildcardPermission explained here](http://shiro.apache.org/permissions.html). The previous documentation for the Spring Security permissions could come in handy for you: https://github.com/stormpath/stormpath-spring-security/wiki#permissions